pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,104 @@
+package telemetry
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"os"
+	"time"
+
+	"net/http"
+	"net/url"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+// eventType
+const eventType = "library-os-ng"
+
+// actionName
+const actionName = "Piper Library OS"
+
+// LibraryRepository that is passed into with -ldflags
+var LibraryRepository string
+
+// SiteID
+var SiteID string
+
+var disabled bool
+var client piperhttp.Sender
+
+// Initialize sets up the base telemetry data and is called in generated part of the steps
+func Initialize(telemetryActive bool, _ func(rootPath, resourceName, parameterName string) string, _, stepName string) {
+	//TODO: change parameter semantic to avoid double negation
+	disabled = !telemetryActive
+
+	// skip if telemetry is dieabled
+	if disabled {
+		return
+	}
+
+	if client == nil {
+		client = &piperhttp.Client{}
+	}
+
+	client.SetOptions(piperhttp.ClientOptions{Timeout: time.Second * 5})
+
+	if len(LibraryRepository) == 0 {
+		LibraryRepository = "https://github.com/n/a"
+	}
+
+	if len(SiteID) == 0 {
+		SiteID = "827e8025-1e21-ae84-c3a3-3f62b70b0130"
+	}
+
+	baseData = BaseData{
+		URL:             LibraryRepository,
+		ActionName:      actionName,
+		EventType:       eventType,
+		StepName:        stepName,
+		SiteID:          SiteID,
+		PipelineURLHash: getPipelineURLHash(), // http://server:port/jenkins/job/foo/
+		BuildURLHash:    getBuildURLHash(),    // http://server:port/jenkins/job/foo/15/
+	}
+	//ToDo: register Logrus Hook
+
+}
+
+func getPipelineURLHash() string {
+	return toSha1(os.Getenv("JOB_URL"))
```

Comment:
@OliverNocon this doesn't seem to work as the hash equals the one generated for `BUILD_URL`. Is it necessary to hand down the env var into the docker?

Reply:
not sure, I guess we need to do some more tests ...

Reply:
I just ran a test and the environment variables are properly available also in a Kubernetes pod.
What is your setup where this does not seem to be the case?

Reply:
Sorry, this showed up when I executed it locally, where no env vars are set. I changed this now to not report a SHA1 in that case but a plain string `n/a`.